[Chan Choon Yong] iP

choonyongchan opened the pull request:
**DukeCY**

> I would like to die on Mars. Just not on impact. - Elon Musk [source](https://i.kym-cdn.com/photos/images/original/001/820/208/0d5.jpg)

DukeCY is a one-size-fits-all task manager (better than the built-in one with Windows!!). It's
* CLI-based
* Dummy-proof
* Faster than light

All you need to do is,
1. download it from [here](https://github.com/choonyongchan/ip/releases/download/untagged-4450915cef4fc0591fc7/DukeCY.jar).
2. double-click it.
3. add your tasks.
4. let Duke ~spy on you~ manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Reminders

If you are a Java programmer, you can use it to practice Java too. Here's the main method.

    public class Duke {
        public static void main(String[] args) {
            /*wait what now*/
        }
    }

Commit:
Add Gradle support

choonyongchan committed:
Committing Level-1

choonyongchan committed:
Committing Level-2

choonyongchan committed:
Commiting Level-3

choonyongchan committed:
Committing Level-4

choonyongchan committed:
Committing Level-5

choonyongchan committed:
Committing Level-6

choonyongchan committed:
Committing A-TextUiTesting

choonyongchan committed:
Add Save and Load Functionality

choonyongchan committed:
Add Date Formatting Function

choonyongchan committed:
Merge branch 'branch-Level-8'

choonyongchan committed:
Format project to OOP

choonyongchan committed:
Separate project into Java packages

choonyongchan committed:
Add JUnit Test

choonyongchan committed:
Add JavaDoc comments

tianyue58 commented on src/main/duke/Storage.java:
```diff
@@ -0,0 +1,77 @@
+        for (int i = 0; i < tasks.size(); i++) {
+            listString += String.format("%s\n", tasks.get(i).saveString());
+        }
+        try (PrintWriter writer = new PrintWriter(file)){
```
Would it be better if you move "PrintWriter writer = new PrintWriter(file)" somewhere else rather than inside () for try block?

tianyue58 commented on src/main/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,58 @@
+            break;
+        case "deadline":
+            String[] deadlineDescriptions = fullCommand.replace("deadline", "").split("/by");
+            if (deadlineDescriptions[0].trim().isEmpty() || deadlineDescriptions[1].trim().isEmpty()) {
```
Would it be better to declare the boolean as a variable on top? You can have a look at the Avoid Complicated Expressions section in this link. 
https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html

tianyue58 commented on src/main/duke/Storage.java:
```diff
@@ -0,0 +1,77 @@
+
+    private File file;
+    
+    public Storage(String filePath) {
```
Don't forget to write header comments for all classes. :)

choonyongchan replied:
Thanks for the reminder. I'll add the comments soon :)

tianyue58 reviewed (commented):
LGTM. Just some small things to fix! 😃

choonyongchan committed:
Improve Coding Standard

choonyongchan committed:
Add Find feature

choonyongchan committed:
Merge branch 'branch-A-CodingStandard'

choonyongchan committed:
Merge branch 'branch-Level-9'

pattan9955 commented on src/main/duke/Parser.java:
```diff
@@ -0,0 +1,39 @@
+public class Parser {
+
+    /**
+     * Returns command according to input response.
```
Should there be a new line between the method description and the arguments?

choonyongchan replied:
Thanks for highlighting this out. I'll make the changes :)

pattan9955 commented on src/main/duke/Parser.java:
```diff
@@ -0,0 +1,39 @@
+        case "bye":
+            return new ExitCommand(fullCommand);
+        case "list":
+            return  new ListCommand(fullCommand);
```
Perhaps there was an accidental addition of a space between "return" and "new"?

pattan9955 commented on src/main/duke/Parser.java:
```diff
@@ -0,0 +1,39 @@
+        String type = fullCommand.split(" ")[0];
+        switch (type) {
+        case "bye":
+            return new ExitCommand(fullCommand);
```
Should the case statements in the switch block be unindented?

pattan9955 commented on src/main/duke/Storage.java:
```diff
@@ -0,0 +1,100 @@
+        Task task;
+        switch (type) {
+            case "T":
+                task = new TodoTask(description);
```
Should the case statements inside the switch block be unindented?

pattan9955 commented on src/main/duke/Ui.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Display divider line between user command and Duke chatbot response.
```
I like how your comments are short, sweet, and straight to the point! 😄

pattan9955 commented on src/main/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Differentate which tasks user is adding.
```
Short and sweet description, however I noticed a similar issue elsewhere, where the description and arguments were missing a line in between.

pattan9955 reviewed (commented):
Overall, LGTM! The code was well written and easy to read, with only a few nits here and there pertaining to some minor coding standard violations. 👍

choonyongchan reviewed (commented):
Thanks both for the helpful comments!

choonyongchan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

choonyongchan committed:
Add Gradle Support

choonyongchan committed:
Add GUI Function

choonyongchan committed:
Add GUI Skeleton Code

choonyongchan committed:
Fixed GUI Support

choonyongchan committed:
Improve Code Quality

choonyongchan committed:
Revert "Improve Code Quality"

This reverts commit 75864223f1a83945037ddbc5424e4c09db455417.

choonyongchan committed:
Improve Code Quality

Make coding style consistent

Coding standard is inconsistent across files.

This may cause confusion when others wish to read my code.

Let's follow the CS2103T coding standard.

This is to ensure conformity with the general coding standard

choonyongchan committed:
Create gradle.yml

choonyongchan committed:
Revert "Improve Code Quality"

This reverts commit c449054aabbfc08d30d74e05a0bcde06b2725c06.

choonyongchan committed:
Revert "Create gradle.yml"

This reverts commit e81e9653871db855b5fac2c83b2569f9000238da.

choonyongchan committed:
Revert "Revert "Revert "Create gradle.yml"""

This reverts commit ce5f344350a4f76375507feecc54d0fe3bd398e9.

choonyongchan committed:
Merge pull request #1 from choonyongchan/branch-Week-5

Revert "Revert "Revert "Create gradle.yml"""

choonyongchan committed:
Add Duplicate Checking Function

choonyongchan committed:
Merge pull request #2 from choonyongchan/branch-Week-5

Add Duplicate Checking Function

choonyongchan committed:
Improve GUI Features

choonyongchan committed:
Set theme jekyll-theme-tactile

choonyongchan committed:
Finalise Features

choonyongchan committed:
Merge branch 'branch-A-BetterGui'

choonyongchan committed:
Fixing JavaFx Errors
